modules/fedimint-mint-client/src/repair_wallet.rs

```diff
@@ -0,0 +1,146 @@
+use std::collections::BTreeMap;
+
+use fedimint_core::db::IDatabaseTransactionOpsCoreTyped;
+use fedimint_core::util::backoff_util::aggressive_backoff;
+use fedimint_core::util::retry;
+use fedimint_core::{Amount, TieredCounts};
+use futures::{StreamExt, TryStreamExt, stream};
+
+use crate::api::MintFederationApi;
+use crate::client_db::{
+    NextECashNoteIndexKey, NextECashNoteIndexKeyPrefix, NoteKey, NoteKeyPrefix,
+};
+use crate::output::NoteIssuanceRequest;
+use crate::{MintClientModule, NoteIndex};
+
+const CHECK_PARALLELISM: usize = 16;
+
+#[derive(Debug, Clone, Default)]
+pub struct RepairSummary {
+    /// Number of e-cash notes that were found to be spent and removed from the
+    /// wallet per denomination
+    pub spent_notes: TieredCounts,
```

Comment:
The fact that these two fields have a different format/type is a bit odd, unless I miss something.